tests/specs/lemmas.k

```diff
@@ -213,17 +229,7 @@ module LEMMAS-HASKELL [symbolic, kore]
     imports EVM
     imports EDSL
 
-    rule WS ++ .ByteArray => WS  [simplification]
-    rule .ByteArray ++ WS => WS  [simplification]
-
-    rule #sizeByteArray(BUF1 ++ BUF2)                  => #sizeByteArray(BUF1) +Int #sizeByteArray(BUF2)  [simplification]
-    rule #sizeByteArray(#buf(SIZE, _))                 => SIZE                                            [simplification]
-    rule #sizeByteArray(#buf(SIZE, _)[START .. WIDTH]) => WIDTH
-      requires #range(0 <= START < SIZE)
-       andBool #range(0 < WIDTH <= SIZE -Int START)                                                       [simplification]
-
-    rule BUF [ L .. W ] => .ByteArray requires W <=Int 0                    [simplification]
-    rule BUF [ 0 .. W ] => BUF        requires W ==Int #sizeByteArray(BUF)  [simplification]
+    rule BUF [ L .. W ] => padRightBytes(.Bytes, W, 0) requires L >=Int #sizeByteArray(BUF)  [simplification]
```

Comment:
I think this rule can be implemented directly in the semantics instead, in `evm-types.md` around line 594.

```
    rule WS [ START .. WIDTH ] => substrBytes(padRightBytes(WS, START +Int WIDTH, 0), START, START +Int WIDTH) requires START <Int #sizeByteArray(WS) [concrete] 
    rule WS [ START .. WIDTH ] => padRightBytes(.Bytes, WIDTH, 0)                                              [owise, concrete] 
```

becomes (just removing `concrete` from the `owise` rule):

```
    rule WS [ START .. WIDTH ] => substrBytes(padRightBytes(WS, START +Int WIDTH, 0), START, START +Int WIDTH) requires START <Int #sizeByteArray(WS) [concrete] 
    rule WS [ START .. WIDTH ] => padRightBytes(.Bytes, WIDTH, 0)                                              [owise] 
```

I think this achieves the same effect, which is to say "if the prover can show that `notBool START <Int #sizeByteArray(WS)`, then simplify to the empty `.Bytes`".

Reply:
ok